### PR TITLE
Enable MySQL CDC + PK 100%

### LINKED ISSUE #30215: source-mysql : enable cdc pk initial load to 100%
Rolling cdc pk initial load to 100% includes
- Enabling cdc pk load by default
- Removing legacy initial snapshot codepath
- Fixing unit + integration tests appropriately

### PR DESCRIPTION
**Pull request by akashkulk**
Closes https://github.com/airbytehq/airbyte/issues/30215 and enables the MySQL CDC initial load for 100%

Fixes the integration tests, removes the legacy CDC snapshot load test

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialReadUtil.java [hunk 1]
```diff
@@ -179,6 +179,7 @@ public static InitialLoadStreams streamsForInitialPrimaryKeyLoad(final CdcStateM
                                                                    final ConfiguredAirbyteCatalog fullCatalog,
                                                                    final boolean savedOffsetStillPresentOnServer) {
     if (!savedOffsetStillPresentOnServer) {
+      LOGGER.info("FULL RESYNC DUE TO SAVED OFFSET ISSUE");
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialReadUtil.java:182]: checking: intentionally left this log?

**Review by rodireich — APPROVED** [review]

**Review by rodireich — APPROVED** [review]

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialReadUtil.java (hunk 1 above)
Comment (akashkulk) [reply airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialReadUtil.java:182]: No, I need to revert this file and `MySqlInitialStateIterator.java`. Thanks for checking!

**Review by prateekmukhedkar — APPROVED** [review]

**Review by nguyenaiden — APPROVED** [review]
+1

File: airbyte-integrations/connectors/source-mysql/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcBinlogsMySqlSourceDatatypeTest.java [hunk 2]
```diff
@@ -60,14 +70,15 @@ protected void postSetup() throws Exception {
     catalog.getStreams().add(dummyTableWithData);
 
     final List<AirbyteMessage> allMessages = super.runRead(catalog);
-    if (allMessages.size() != 2) {
+    /*if (allMessages.size() != 2) {
```
Comment (subodh1810) [inline airbyte-integrations/connectors/source-mysql/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/CdcBinlogsMySqlSourceDatatypeTest.java:73]: let's delete this

**Review by subodh1810 — APPROVED** [review]
Some NIT comments 

MySqlCdcStateHandler class method saveStateAfterCompletionOfSnapshotOfNewStreams should now throw runtime exception cause debezium is no longer used for new tables snapshot and this method should never be triggered. 
 MySqlCdcProperties class method getSnapshotProperties is no longer used so should be deleted.

**Comment by octavia-approvington** [comment]
Lets merge it <br/> sheep thumbs up

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**